google/cloud/internal/curl_rest_client_integration_test.cc

```diff
@@ -443,6 +466,31 @@ TEST_F(RestClientIntegrationTest, PeerPseudoHeader) {
   EXPECT_THAT(payload->DebugHeaders(), ContainsOnce(Pair(":curl-peer", _)));
 }
 
+TEST_F(RestClientIntegrationTest, RestContextHeaders) {
+  auto client = MakeDefaultRestClient(url_, {});
+  RestRequest request;
+  request.SetPath("anything");
+  auto response_status = RetryRestRequest([&] {
+    rest_internal::RestContext context;
+    context.AddHeader({"x-test-header-1", "header-value-1"});
+    context.AddHeader({"x-test-header-2", "header-value-2"});
+    return client->Get(context, request);
+  });
+  ASSERT_STATUS_OK(response_status);
+  auto response = *std::move(response_status);
+  ASSERT_THAT(response->StatusCode(), Eq(HttpStatusCode::kOk));
+  auto body = ReadAll(std::move(*response).ExtractPayload());
+  ASSERT_STATUS_OK(body);
+  auto parsed_response = nlohmann::json::parse(*body, nullptr, false);
+  ASSERT_TRUE(parsed_response.is_object());
+  auto sent_headers = parsed_response.find("headers");
+  ASSERT_FALSE(sent_headers == parsed_response.end());
```

Comment:
Opt: `ASSERT_TRUE(... != ...)` might be clearer.

Reply:
Fixed. I also took the opportunity to improve the error messages.